Visual tweaks to vertical navigation

rebeccaalpert opened the pull request:
Adjusted background colors, borders, hover states, icons, and text color, size, and weight on nav.

I didn't address anything under the headline "Navigation Order" on the [design sheet](http://openshift.github.io/openshift-origin-design/web-console/4.0-designs/navigation/navigation#design) since the Jira task seemed to be about styling updates. Let me know if I should include that bit.

Poor-quality gif:
![navbar](https://user-images.githubusercontent.com/7014965/46832423-faf39280-cd73-11e8-9b63-527bc68bc90c.gif)

Closed:
<img width="1346" alt="screen shot 2018-10-11 at 4 23 55 pm" src="https://user-images.githubusercontent.com/7014965/46831708-22e1f680-cd72-11e8-9a0a-0126e91f9669.png">

Open:
<img width="1346" alt="screen shot 2018-10-11 at 4 24 09 pm" src="https://user-images.githubusercontent.com/7014965/46831717-27a6aa80-cd72-11e8-928b-c1380ca567ea.png">

Fixes [CONSOLE-900](https://jira.coreos.com/browse/CONSOLE-900?filter=-1).

@openshift/team-ux-review, can you please take a look? Thanks!

spadgett commented:
/assign @rhamilto

spadgett commented:
jenkins rebuild

spadgett commented on frontend/public/components/nav.jsx:
```diff
@@ -379,7 +376,7 @@ export class Nav extends React.Component {
           </NavSection>
 
-          <NavSection required={FLAGS.OPERATOR_LIFECYCLE_MANAGER} text="Operators" img={operatorImg} activeImg={operatorActiveImg} >
+          <NavSection required={FLAGS.OPERATOR_LIFECYCLE_MANAGER} text="Operators" icon="pficon pficon-on-running">
```
We shouldn't be changing this... This is the OLM logo.

rhamilto commented on frontend/public/components/nav.jsx:
```diff
@@ -14,10 +14,6 @@ import { authSvc } from '../module/auth';
 
-import * as operatorImg from '../imgs/operator.svg';
-import * as operatorActiveImg from '../imgs/operator-active.svg';
-import * as routingImg from '../imgs/routing.svg';
```
Should also delete the routing images.

rhamilto commented on frontend/public/components/_nav.scss:
```diff
@@ -81,10 +83,10 @@ $active-border: 3px;
   cursor: pointer;
   display: flex;
-  font-size: 15px;
+  font-size: 14px;
```
Lines 96-100 below are no longer needed since the routing image is being replaced by an icon.

rebeccaalpert replied:
Design team isn't moving forward with the icons right now, so I reverted those changes.

rhamilto commented on frontend/public/components/_nav.scss:
```diff
@@ -7,24 +7,25 @@ $active-border: 3px;
 
   a {
     background-color: inherit;
-    color: $color-os-nav-title;
```
`$color-os-nav-title` is set to `$color-pf-black-300`.  No change needed.

rhamilto commented on frontend/public/components/_nav.scss:
```diff
@@ -33,7 +34,7 @@ $active-border: 3px;
 
 .co-m-nav-link {
   &:hover {
-    background-color: $color-os-nav-background-active;
```
Update the value of `$color-os-nav-background-active` to `$color-pf-black-700` so as not to orphan the var and require multiple changes.

rhamilto commented on frontend/public/components/_nav.scss:
```diff
@@ -52,27 +53,28 @@ $active-border: 3px;
+}
+
 .navigation-container__section--active {
-  background-color: $color-os-nav-background-active;
```
See earlier comment about `$color-os-nav-background-active`

rhamilto commented on frontend/public/components/_nav.scss:
```diff
@@ -81,10 +83,10 @@ $active-border: 3px;
 
 .navigation-container__section__title {
   align-items: center;
-  color: $color-os-nav-title;
```
See earlier comment about `$color-os-nav-title`

rhamilto commented on frontend/public/components/_nav.scss:
```diff
@@ -112,7 +127,7 @@ $active-border: 3px;
 }
 
 .navigation-container__section__title__icon {
-  color: $color-os-nav-icon;
```
`$color-os-nav-icon` is already set to `$color-pf-black-600`.  No change needed.

rhamilto commented on frontend/public/components/_nav.scss:
```diff
@@ -7,24 +7,25 @@ $active-border: 3px;
+    font-weight: 600;
     padding-left: $link-padding-left - $active-border;
-    background-color: #383F47;
-    border-left: $active-border solid $color-os-nav-border-left-active;
```
`$color-os-nav-border-left-active` is set to `$color-pf-blue-300`.  No change needed.

rhamilto commented on frontend/public/components/_nav.scss:
```diff
@@ -52,27 +53,28 @@ $active-border: 3px;
-  background-color: $color-os-nav-background-active;
+  background-color: $color-pf-black-700;
   position: relative;
-  &:before {
```
We currently don't have nav items without children.  So either this entire block (lines 66-76) and [supporting jsx](https://github.com/openshift/console/blob/master/frontend/public/components/nav.jsx#L225) should go or we should keep the entire rule.

rhamilto commented on frontend/public/components/_nav.scss:
```diff
@@ -52,27 +53,28 @@ $active-border: 3px;
 
-.navigation-container__section, .navigation-container__section__separator {
+.navigation-container__section {
   border-bottom: 1px solid $color-os-nav-item-seperator;
```
The var name doesn't align with the class name.  `$color-os-nav-item-seperator` should go with `.navigation-container__section__separator`.  Adjust accordingly.

rhamilto commented on frontend/public/style/_layout.scss:
```diff
@@ -27,7 +27,7 @@ body,
 }
 
 #sidebar {
-  background-color: $color-os-nav-background;
```
`$color-os-nav-background` is set to `$color-pf-black-900`.  No change needed.

rebeccaalpert commented:
I'll adjust the routing images/rules and the OLM icon once the interaction designers and Sam have discussed whether to use the svg or Patternfly icon for Operators. I addressed the other feedback. Also fixed the spelling on one of the rules ("seperator" vs. "separator") and updated the non-nav file that was using it.

rebeccaalpert commented:
Reverted icon-related changes since the design team isn't moving forward with the icons right now. Also added some additional visual changes (different background color for inactive open sections, arrow on hover, arrow on open sections.)

![navbar](https://user-images.githubusercontent.com/7014965/46882116-87a55b80-ce1b-11e8-9638-acf8970ad31e.gif)

spadgett commented:
@beanh66 @serenamarie125 I just wanted to check when the section should have the lighter background. There are several different states:

* Open, no selection, no hover
* Open, no selection, hover
* Open, item selected
* Closed, no selection, hover
* Closed, item selected

serenamarie125 commented:
@maryshak1996 can you answer sam's questions re: the lighter background?  Thanks!

maryshak1996 commented:
@spadgett 
The following should have the lighter background for the particular nav item
- Open, on a selected page, item on hover
<img width="273" alt="screen shot 2018-10-12 at 4 09 37 pm" src="https://user-images.githubusercontent.com/19824574/46891886-7c612880-ce39-11e8-8c71-3a299308ac83.png">
- Open, item selected
<img width="352" alt="screen shot 2018-10-12 at 4 03 51 pm" src="https://user-images.githubusercontent.com/19824574/46891895-85ea9080-ce39-11e8-900d-98e3f633476b.png">
- Open no selection, primary item on hover
<img width="269" alt="screen shot 2018-10-12 at 4 12 19 pm" src="https://user-images.githubusercontent.com/19824574/46891927-9e5aab00-ce39-11e8-89d9-a74205b5ab56.png">
- Open, no selection, secondary item on hover
<img width="382" alt="screen shot 2018-10-12 at 4 06 10 pm" src="https://user-images.githubusercontent.com/19824574/46891931-a31f5f00-ce39-11e8-8fe6-bab7fc1df097.png">
- Closed, no selection, hover
<img width="256" alt="screen shot 2018-10-12 at 4 12 55 pm" src="https://user-images.githubusercontent.com/19824574/46891989-d530c100-ce39-11e8-867e-f0e2a866c889.png">
- Closed, secondary item selected
<img width="269" alt="screen shot 2018-10-12 at 4 13 25 pm" src="https://user-images.githubusercontent.com/19824574/46891982-cf3ae000-ce39-11e8-853e-5d24ad847032.png">

Does this help? There are so many states!

rebeccaalpert commented:
@maryshak1996 Thanks, the primary item hover states weren't in the design doc. Should the case open selection, primary item on hover have a hover state?

(i.e. if you had your mouse on "Workloads", would it look like this?):
<img width="215" alt="screen shot 2018-10-12 at 11 15 26 am" src="https://user-images.githubusercontent.com/7014965/46892627-1629d500-ce3c-11e8-9f91-fb68fb903071.png">

maryshak1996 commented:
@rebeccaalpert in the case of the nav being expanded and primary item being hovered over, here is the treatment (that primary item gets the hover effect, not a subnav item). 
![screen shot 2018-10-15 at 9 16 24 am](https://user-images.githubusercontent.com/19824574/46953298-7cd90980-d05b-11e8-9463-03d5b7f85fda.png)

Also, I don't see the carets on the nav primary items @serenamarie125 is that something that we aren't doing anymore? I had thought that we had agreed to have them?

serenamarie125 commented:
@maryshak1996 the final decision was to have the carets show up on hover, similar to what we are doing in the new Overview, to minimize the noise.

rebeccaalpert commented:
Here's an updated gif showing the interactions (with primary item hover states):
![navbar-updated](https://user-images.githubusercontent.com/7014965/46958147-207be700-d067-11e8-9b30-2c07277af723.gif)

rebeccaalpert commented:
Here's a gif showing the updated states (including the section title hover states):
![navbar-updated](https://user-images.githubusercontent.com/7014965/46958338-9d0ec580-d067-11e8-84fb-fde8e54f22a8.gif)

maryshak1996 commented:
Looks good to me!

rhamilto commented on frontend/public/components/_nav.scss:
```diff
@@ -52,39 +53,39 @@ $active-border: 3px;
 }
 
+.navigation-container--open {
+  .navigation-container__list {
```
BEM suggests the avoidance of nesting.  The modifier should go on the child, not the parent.  Here and below.

Nit:  alphabetize rules by selector.  Here and below.

rhamilto commented on frontend/public/components/_nav.scss:
```diff
@@ -52,39 +53,39 @@ $active-border: 3px;
+  .navigation-container__list {
+    margin: 5px 0 10px 0;
+  }
+  .navigation-container__item__separator {
```
BEM says don't nest elements.  This should be `.navigation-container__separator`.  Here and below.  I realize you're following the existing code, but we should probably clean them all up.

tlwu2013 commented:
I notice the `.active` state pushes the `navigation-container__section__title` toward the right and causes the misalignment:
![screenshot-1](https://user-images.githubusercontent.com/5903705/47057283-cb2dfb80-d174-11e8-9925-01a3647fa0e8.png)

but not for the `co-m-nav-link`:
![screenshot-2](https://user-images.githubusercontent.com/5903705/47057364-2102a380-d175-11e8-8938-7c88b5a0691a.png)

@rebeccaalpert  could you help look into this as well?  thanks!

rebeccaalpert commented:
Fixed the active title padding:
<img width="218" alt="screen shot 2018-10-17 at 10 04 02 am" src="https://user-images.githubusercontent.com/7014965/47092007-07ed0780-d1f4-11e8-8bde-0f514fdbd952.png">

Updated BEM name (didn't adjust separator modifier since it's a separate const used in a different component) and alphabetization in a couple places.

rebeccaalpert commented:
jenkins rebuild

spadgett commented on frontend/public/components/_nav.scss:
```diff
@@ -52,39 +53,40 @@ $active-border: 3px;
 }
 
+.navigation-container--open {
+  .navigation-container__separator {
```
nit:

```css
.navigation-container--open .navigation-container__separator {
```

spadgett commented on frontend/public/components/_nav.scss:
```diff
@@ -101,7 +103,36 @@ $active-border: 3px;
 }
+
+.navigation-container__section__title--inactive-open {
+  &:hover {
```
```css
.navigation-container__section__title--inactive-open:hover {
```

spadgett commented on frontend/public/components/nav.jsx:
```diff
@@ -238,22 +238,25 @@ const NavSection = connect(navSectionStateToProps)(
 
-      return <div className={classNames(sectionClassName, klass)}>
-        <div id={id} className="navigation-container__section__title" onClick={this.toggle}>
+      return <div className={classNames('navigation-container__section', klass, {'navigation-container--active': isActive}, {'navigation-container--open': isOpen})}>
```
```suggestion
      return <div className={classNames('navigation-container__section', klass, {'navigation-container--active': isActive, 'navigation-container--open': isOpen})}>
```

spadgett commented on frontend/public/components/nav.jsx:
```diff
@@ -238,22 +238,25 @@ const NavSection = connect(navSectionStateToProps)(
           }
+          {isOpen
+            ? <i className="icon fa fa-angle-down navigation-container__section__title__icon" aria-hidden="true"></i>
+            : <i className="icon fa fa-angle-right navigation-container__section__title__icon" aria-hidden="true"></i>}
```
```js
<i className={classNames('icon navigation-container__section__title__icon', isOpen ? 'fa-angle-down' : 'fa-angle-right')} aria-hidden="true" />
```

spadgett commented:
LGTM, please squash. Thanks!

rebeccaalpert commented:
Squashed!

spadgett commented:
Hm, linter error:

```
/go/src/github.com/openshift/console/frontend/public/components/nav.jsx
  242:143  error  Duplicate key 'navigation-container__section__title--active'  no-dupe-keys
```

rebeccaalpert commented:
I did; I took a partial day yesterday. I'll look into this today.

rebeccaalpert commented:
Quick fix. Had a duplicate class in there due to a copy/paste fail.

rebeccaalpert commented:
jenkins rebuild

spadgett commented:
/lgtm

spadgett commented:
jenkins rebuild

spadgett commented:
jenkins rebuild

rebeccaalpert commented:
jenkins rebuild

rebeccaalpert commented:
jenkins rebuild

spadgett commented:
I'm hoping #694 fixes the OLM failure

rebeccaalpert commented:
jenkins rebuild

rebeccaalpert commented:
jenkins rebuild

spadgett commented:
jenkins rebuild

spadgett commented:
/lgtm
